account_payment_order/models/account_payment.py

```diff
@@ -87,3 +89,28 @@ def _prepare_move_line_default_vals(self, write_off_line_vals=None):
         for vals in vals_list:
             vals["date_maturity"] = self.payment_line_ids[0].date
         return vals_list
+
+    @api.depends(
+        "partner_id",
+        "company_id",
+        "payment_type",
+        "destination_journal_id",
+        "is_internal_transfer",
+    )
+    def _compute_available_partner_bank_ids(self):
+        original_payment_ids = self.filtered(
+            lambda pay: pay.payment_method_code not in OCA_PAYMENT_METHOD_CODES
+        )
+        super(
+            AccountPayment, original_payment_ids
+        )._compute_available_partner_bank_ids()
+        oca_payment_ids = self - original_payment_ids
+        for pay in oca_payment_ids:
+            if pay.payment_method_code == "sepa_direct_debit":
+                pay.available_partner_bank_ids = pay.partner_id.bank_ids.filtered(
+                    lambda x: x.company_id.id in (False, pay.company_id.id)
+                )._origin
+            if pay.payment_method_code == "sepa_credit_transfer":
+                pay.available_partner_bank_ids = pay.partner_id.bank_ids.filtered(
+                    lambda x: x.company_id.id in (False, pay.company_id.id)
+                )._origin
```

Comment:
```python
        return res
```

Reply:
thanks for the review @MiquelRForgeFlow 
I'm wondering what's the interest of returning something even if  the original method is not returning anything ?

Regards